src/entities/mellow/mellowLpRouter.ts

```diff
@@ -633,6 +662,95 @@ class MellowLpRouter {
       throw new Error('Unsucessful rollover confirmation.');
     }
   };
+
+  registerForAutoRollover = async (registration: boolean): Promise<ContractReceipt> => {
+    if (
+      isUndefined(this.readOnlyContracts) ||
+      isUndefined(this.writeContracts) ||
+      isUndefined(this.userAddress)
+    ) {
+      throw new Error('Uninitialized contracts.');
+    }
+
+    try {
+      await this.writeContracts.mellowRouter.callStatic.registerForAutoRollover(registration);
+    } catch (err) {
+      sentryTracker.captureException(err);
+      sentryTracker.captureMessage('Unsuccessful auto-rollover registration simulation');
+      console.error('Error during registration for auto-rollover', err);
+      throw new Error('Unsuccessful auto-rollover registration simulation');
+    }
+
+    const gasLimit = await this.writeContracts.mellowRouter.estimateGas.registerForAutoRollover(
+      registration,
+    );
+
+    const tx = await this.writeContracts.mellowRouter.registerForAutoRollover(registration, {
+      gasLimit: getGasBuffer(gasLimit),
+    });
+
+    try {
+      const receipt = await tx.wait();
+
+      return receipt;
+    } catch (err) {
+      sentryTracker.captureException(err);
+      sentryTracker.captureMessage('Unsucessful auto-rollover registration confirmation.');
+      throw new Error('Unsucessful auto-rollover registration confirmation.');
+    }
+  };
+
+  autorolloverRegistrationFee = async (registration: boolean): Promise<string> => {
+    if (
+      isUndefined(this.readOnlyContracts) ||
+      isUndefined(this.writeContracts) ||
+      isUndefined(this.userAddress)
+    ) {
+      throw new Error('Uninitialized contracts.');
+    }
+
+    try {
+      await this.writeContracts.mellowRouter.callStatic.registerForAutoRollover(registration);
+    } catch (err) {
+      sentryTracker.captureException(err);
+      sentryTracker.captureMessage('Unsuccessful auto-rollover registration simulation');
+      console.error('Error during registration for auto-rollover', err);
+      throw new Error('Unsuccessful auto-rollover registration simulation');
+    }
+
+    // returns gas estimate in gas units
+    const gasUnitsEstimate =
+      await this.writeContracts.mellowRouter.estimateGas.registerForAutoRollover(registration);
+
+    // convert gas estimate from gas units into usd
+    const ethToUSDPrice = await geckoEthToUsd();
+    const gasPriceGwei = await getGasPriceGwei();
+    const gasPriceUSD = convertGasUnitsToUSD(gasUnitsEstimate, ethToUSDPrice, gasPriceGwei);
+    return gasPriceUSD;
+  };
+
+  getAutorolloverRegistrationFlag = async (userAddress: string): Promise<boolean> => {
+    if (
+      isUndefined(this.readOnlyContracts) ||
+      isUndefined(this.writeContracts) ||
```

Comment:
we don't need to check this.writeContract for this function.

Reply:
resolved